### PR TITLE
503 for bad db connection

### PR DESCRIPTION
**Pull request by defong**
### Context
bad db connection

### Changes proposed in this pull request
Added config for bad connection to be http status 503

### Guidance to review
Using production config, hence no stack trace

before 
image

after
image

### CODE REVIEW

**Comment by defong** [comment]
[UCAS API error handling](https://trello.com/c/59vGAAoM/875-ucas-api-error-handling)

File: spec/controllers/api/v1/courses_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::CoursesController, type: :controller do
+  before do
+    controller.request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Token.encode_credentials('bats')
+    def controller.index
```
Comment (misaka) [inline spec/controllers/api/v1/courses_controller_spec.rb]: Use:

```
allow(controller).to received(:index).and_raise(PG:ConnectionBad)
```

and do this in the spec where you expect to raise.

File: spec/controllers/api/v1/providers_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::ProvidersController, type: :controller do
+  before do
+    controller.request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Token.encode_credentials('bats')
+    def controller.index
```
Comment (misaka) [inline spec/controllers/api/v1/providers_controller_spec.rb]: Use:

```
allow(controller).to received(:index).and_raise(PG:ConnectionBad)
```

and do this in the spec where you expect to raise.

File: spec/controllers/api/v1/subjects_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::SubjectsController, type: :controller do
+  before do
+    controller.request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Token.encode_credentials('bats')
+    def controller.index
```
Comment (misaka) [inline spec/controllers/api/v1/subjects_controller_spec.rb]: Use:

```
allow(controller).to received(:index).and_raise(PG:ConnectionBad)
```

and do this in the spec where you expect to raise.

File: spec/controllers/api/v1/courses_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  describe "index" do
+    it "render service unavailable" do
+      allow(controller).to receive(:index).and_raise(PG::ConnectionBad)
+      allow(controller).to receive(:authenticate)
+
+      get :index
+      expect(response).to have_http_status(:service_unavailable)
+      json = JSON.parse(response.body)
+      expect(json). to eq(
+        'code' => 503, 'status' => 'Service Unavailable'
```
Comment (dankmitchell) [inline spec/controllers/api/v1/courses_controller_spec.rb]: Can this be written as `code: 503, status: 'Service Unavailable'` ?

Fat arrows went out of fashion in Ruby 1.8

Comment (defong) [reply spec/controllers/api/v1/courses_controller_spec.rb]: argh, now i got to contend with catwalks and  fashion sense as well

:cry:

Comment (dankmitchell) [reply spec/controllers/api/v1/courses_controller_spec.rb]: Not a blocker. More of an FYI ;-)

**Pull request merged**